experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -470,23 +464,146 @@ export interface Sequence<TTypes extends AllowedTypes> extends TreeField {
 	at(index: number): UnboxNodeUnion<TTypes>;
 
 	/**
-	 * Gets a boxed node of this field by its index.
-	 * Note that a node must exist at the given index.
+	 * Returns the index of the first occurrence of a value in the sequence, or -1 if it is not present.
+	 * @param searchElement - The element to search for.
+	 * @param fromIndex - The index at which to begin the search. If fromIndex is omitted, the search starts at index 0.
 	 */
-	boxedAt(index: number): TypedNodeUnion<TTypes>;
+	indexOf(searchElement: UnboxNodeUnion<TTypes>, fromIndex?: number): number;
 
 	/**
-	 * Calls the provided callback function on each child of this sequence, and returns an array that contains the results.
-	 * @param callbackfn - A function that accepts the child, its index, and this field.
+	 * Determines whether an sequence includes a certain element, returning true or false as appropriate.
+	 * @param searchElement - The element to search for.
+	 * @param fromIndex - The index at which to begin the search. If fromIndex is omitted, the search starts at index 0.
+	 */
+	includes(searchElement: UnboxNodeUnion<TTypes>, fromIndex?: number): boolean;
+
+	/**
+	 * Iterates the indices of the elements in the sequence.
+	 */
+	keys(): IterableIterator<number>;
+
+	/**
+	 * Iterates the values in the sequence.
+	 */
+	values(): IterableIterator<UnboxNodeUnion<TTypes>>;
+
+	/**
+	 * Iterates the elements in the sequence as tuples of index and element.
 	 */
-	map<U>(callbackfn: (value: UnboxNodeUnion<TTypes>, index: number, array: this) => U): U[];
+	entries(): IterableIterator<[number, UnboxNodeUnion<TTypes>]>;
+
+	/**
+	 * Executes a provided function once per each element in the sequence.
+	 * @param callbackFn - The function to run for each element.
+	 */
+	forEach(
+		callbackFn: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => void,
+	): void;
+
+	/**
+	 * Returns the value of the first element in the sequence where `predicate` is true, and undefined otherwise.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	find(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): UnboxNodeUnion<TTypes> | undefined;
+	find<T extends UnboxNodeUnion<TTypes>>(
+		predicate: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => element is T,
+	): T | undefined;
+
+	/**
+	 * Returns the value of the last element in the sequence where `predicate` is true, and undefined otherwise.
```

Comment:
THis seems like a bad idea. Since it unboxes, undefined is a valid value that could be in the list, and people may inccorrectly think something was not found when you did find undefined.

Reply:
@CraigMacomber - My understanding is that the only way to get the 'undefined' value is to read a non-existent optional field.  Lists contain nodes, not fields, and therefore cannot be undefined?